addon/services/segment.js

```diff
@@ -1,88 +1,148 @@
 /* globals FastBoot */
-import Ember from 'ember';
+import Service from '@ember/service';
+import {warn} from '@ember/debug';
 
-export default Ember.Service.extend({
-  init: function() {
+export default Service.extend({
+  _disabled: false,
+  _defaultPageTrackDisabled: false,
+  _defaultIdentifyUserDisabled: false,
+
+  _calledPageTrack: false,
+
+  init: function () {
     this._super();
 
     const isFastBoot = typeof FastBoot !== 'undefined';
 
     if (!this.hasAnalytics() && (this.config && this.config.environment !== 'test') && !isFastBoot) {
-      Ember.Logger.warn('Segment.io is not loaded yet (window.analytics)');
+      warn('Segment.io is not loaded yet (window.analytics)', false, {
+        id: 'ember-cli-segment.analytics-not-loaded'
+      });
     }
+
+    const hasSegmentConfig = (this.config && this.config.segment);
+    this.set('_defaultPageTrackDisabled', hasSegmentConfig && this.config.segment.defaultPageTrack === false);
+    this.set('_defaultIdentifyUserDisabled', hasSegmentConfig && this.config.segment.defaultIdentifyUser === false)
+    this.set('_disabled', hasSegmentConfig && this.config.segment.enabled === false)
   },
 
-  hasAnalytics: function() {
+  hasAnalytics: function () {
     return !!(window.analytics && typeof window.analytics === "object");
   },
 
-  // Default true unless user explicitly sets defaultPageTrack to false
-  pageTrackEnabled: function() {
+  enabled: function () {
+    return !this.disabled();
+  },
+
+  disabled: function () {
```

Comment:
Do you mind naming this as `isDisabled` ?

Reply:
I don't mind. Should I rename `pageTrackEnabled/pageTrackDisabled` as well or leave them as is?